src/rebuild/srv.c

```diff
@@ -1310,36 +1364,56 @@ ds_rebuild_leader_stop_all()
 		ABT_cond_free(&rebuild_gst.rg_stop_cond);
 }
 
+static void
+rebuild_print_list_update(const char *const str, const uuid_t uuid,
+			  const uint32_t map_ver,
+			  daos_rebuild_opc_t rebuild_op,
+			  struct pool_target_id_list *tgts) {
+	int i;
+
+	D_PRINT("%s (pool="DF_UUID" ver=%u, op=%s) tgts=", str, DP_UUID(uuid),
+		map_ver, RB_OP_STR(rebuild_op));
+	for (i = 0; i < tgts->pti_number; i++) {
+		if (i > 0)
+			D_PRINT(",");
+		D_PRINT("%u", tgts->pti_ids[i].pti_id);
+	}
+	D_PRINT("\n");
+}
+
 /**
  * Add rebuild task to the rebuild list and another ULT will rebuild the
  * pool.
  */
 int
 ds_rebuild_schedule(const uuid_t uuid, uint32_t map_ver,
-		    struct pool_target_id_list *tgts_failed)
+		    struct pool_target_id_list *tgts,
+		    daos_rebuild_opc_t rebuild_op)
 {
 	struct rebuild_task	*task;
 	int			rc;
 
 	/* Check if the pool already in the queue list */
-	rc = rebuild_try_merge_tgts(uuid, map_ver, tgts_failed);
+	rc = rebuild_try_merge_tgts(uuid, map_ver, rebuild_op, tgts);
 	if (rc)
 		return rc == 1 ? 0 : rc;
 
+	/* No existing task was found - allocate a new one and use it */
 	D_ALLOC_PTR(task);
 	if (task == NULL)
 		return -DER_NOMEM;
 
 	task->dst_map_ver = map_ver;
+	task->dst_rebuild_op = rebuild_op;
 	uuid_copy(task->dst_pool_uuid, uuid);
 	D_INIT_LIST_HEAD(&task->dst_list);
 
-	rc = pool_target_id_list_merge(&task->dst_tgts, tgts_failed);
+	rc = pool_target_id_list_merge(&task->dst_tgts, tgts);
 	if (rc)
 		D_GOTO(free, rc);
 
-	D_PRINT("Rebuild [queued] ("DF_UUID" ver=%u) id %u\n",
-		 DP_UUID(uuid), map_ver, tgts_failed->pti_ids[0].pti_id);
+	rebuild_print_list_update("Rebuild queued",
+				  uuid, map_ver, rebuild_op, tgts);
 	d_list_add_tail(&task->dst_list, &rebuild_gst.rg_queue_list);
```

Comment:
probably need separate rebuild and re-integration list, so it would be easier to favorite rebuild job first, then re-integration.

Reply:
Agreed, but I'd like to do this in the next patch.